storage/migrations/01_consensus.up.sql

```diff
@@ -288,11 +288,16 @@ CREATE TABLE chain.processed_blocks
 (
   height         UINT63 NOT NULL,
   analyzer       TEXT NOT NULL,
-  processed_time TIMESTAMP WITH TIME ZONE NOT NULL,
+
+  processed_time TIMESTAMP WITH TIME ZONE, -- NULL if the block is not yet processed.
+  locked_time     TIMESTAMP WITH TIME ZONE, -- NULL if the block is not locked.
```

Comment:
```suggestion
  locked_time    TIMESTAMP WITH TIME ZONE NOT NULL, -- Lock is considered expired if it is old enough; interpretation is up to the analyzer.
```
At least this is the current implementation.

Reply:
The `UnlockBlockForProcessing` can set it to `NULL`  (fast-unlock in case of block process failure).

However, we could change the `NULL` there to be `-infinity` (or `0`). Not sure why I didn't think of this before. I'll change it, so I can remove the `OR IS NULL` check in at least one place.